spharpy/spherical.py

```diff
@@ -1,104 +1,585 @@
+        return self._basis_inv_gradient_theta
+
+    @property
+    def basis_inv_gradient_phi(self):
+        if (
+            self._basis_inv_gradient_theta is None
+            or self._basis_inv_gradient_phi is None
+        ):
+            self.compute_inverse_gradient()
+        self._check_properties(compute='basis_inv_gradient')
+        return self._basis_inv_gradient_phi
+
+    def compute_inverse_gradient(self):
+        if self.inverse_transform is None:
+            ValueError("Inverse transform type not specified")
+        elif self.inverse_transform == "pseudo_inverse":
+            self._basis_inv_gradient_theta = np.linalg.pinv(
+                self.basis_gradient_theta)
+            self._basis_inv_gradient_phi = np.linalg.pinv(
+                self.basis_gradient_phi)
+        elif self.inverse_transform == "quadrature":
+            if self.weights is None:
+                print(
+                    "Warning: No weights specified for quadrature transform,"
+                    "calculating weights using voronoi tessellation of sphere"
+                )
+                self.weights = calculate_sampling_weights(self.coords)
+            self._basis_inv_gradient_theta = np.conj(
+                self.basis_gradient_theta).T * (4 * np.pi * self.weights)
+            self._basis_inv_gradient_phi = np.conj(
+                self.basis_gradient_phi).T * (4 * np.pi * self.weights)
+
+    @basis.setter
+    def basis(self, value):
+        self._basis = value
+
+    @basis_gradient_theta.setter
+    def basis_gradient_theta(self, value):
+        self._basis_gradient_theta = value
+
+    @basis_gradient_phi.setter
+    def basis_gradient_phi(self, value):
+        self._basis_gradient_phi = value
+
+    def _check_properties(self, compute=None):
+        # Check if any crucial properties have changed
+        if (self.n_max != self._prev_n_max or
+                self.coords != self._prev_coords or
+                self.basis_type != self._prev_basis_type or
+                self.inverse_transform != self._prev_inverse_transform or
+                self.channel_convention != self._prev_channel_convention or
+                self.normalization != self._prev_normalization):
+            if compute == 'basis':
+                self._compute_basis()
+            elif compute == 'basis_gradient':
+                self._compute_basis_gradient()
+            elif compute == 'basis_inv':
+                self.compute_inverse()
+            elif compute == 'basis_inv_gradient':
+                self.compute_inverse_gradient()
+            # And update the previous property values
+            self._prev_n_max = self.n_max
+            self._prev_coords = self.coords
+            self._prev_basis_type = self.basis_type
+            self._prev_inverse_transform = self.inverse_transform
+            self._prev_channel_convention = self.channel_convention
+            self._prev_normalization = self.normalization
```

Comment:
Sorry, I think I missed the point of this method. What does this do or what is the benefit of this method?